src/components/LetterSelection/index.jsx

```diff
@@ -1,9 +1,130 @@
+      if (
+        !state.clickedLetterIndexes.includes(action.payload.index) &&
+        state.letterCounter < state.clickedLetters.length
+      ) {
+        return {
+          ...state,
+          letterCounter: state.letterCounter + 1,
+          clickedLetters: state.clickedLetters.map((letter, index) => {
+            if (index === state.letterCounter) {
+              letter = action.payload.letter;
+            }
+            return letter;
+          }),
+          clickedLetterIndexes: [
+            ...state.clickedLetterIndexes,
+            action.payload.index,
+          ],
+        };
+      } else {
+        return state;
+      }
+
+    case types.removeLetter:
+      const clickedLetterIndexesPopped = state.clickedLetterIndexes.slice(
+        0,
+        -1
+      );
+      return {
+        ...state,
+        letterCounter: state.letterCounter === 0 ? 0 : state.letterCounter - 1,
+        clickedLetters: state.clickedLetters.map((letter, index) => {
+          if (index + 1 === state.letterCounter) {
+            letter = '';
+          }
+          return letter;
+        }),
+        clickedLetterIndexes: clickedLetterIndexesPopped,
+      };
+
+    case types.initialize: // initalize the state with resetting letterCounter, new selectable letters, and new blank strings
+      return {
+        ...state,
+        letterCounter: 0,
+        selectableLetters: makeSelectableLetters(action.payload),
+        clickedLetters: makeBlankLetters(action.payload),
+        clickedLetterIndexes: [],
+      };
+
+    default:
+      return state;
+  }
+};
+
+const initialLetterState = {
+  letterCounter: 0,
+  selectableLetters: [],
+  clickedLetters: [], // keeps track of letters that were clicked
+  clickedLetterIndexes: [], // keeps track of letter indexes that were clicked
+};
+
+export default function LetterSelection({ word, dispatchGame, typesGame }) {
+  const [letterState, dispatch] = useReducer(reducer, initialLetterState);
+
+  useEffect(() => {
+    dispatch({
+      type: types.initialize,
+      payload: word,
+    });
+  }, [word]);
 
-export default function LetterSelection() {
   return (
     <>
-      <h2>Letters to choose</h2>
+      {letterState.clickedLetters.map((clickedLetter, index) => {
+        return <div key={index}>{clickedLetter}</div>;
+      })}
+
+      {/* Buttons to select the letters */}
+      {letterState.selectableLetters.map((selectableLetter, index) => {
+        return (
+          <button
+            key={index}
+            onClick={() =>
+              dispatch({
+                type: types.addLetter,
+                payload: { index: index, letter: selectableLetter },
+              })
+            }
+          >
+            {selectableLetter}
+          </button>
+        );
+      })}
+      <button onClick={() => dispatch({ type: types.removeLetter })}>
+        Backspace
+      </button>
+      <button
+        onClick={() =>
+          dispatchGame({
```

Comment:
It is `dispatchGame` because this dispatch is actually the dispatch passed down from the Game component as opposed to the dispatch in the LetterSelection.

Reply:
I see, that makes sense then.